lib/api/core/storage/indexStorage.js

```diff
@@ -0,0 +1,145 @@
+/*
```

Comment:
This class is a wrapper around an index and a storageEngine. It provide the same capabilities than a storage engine but restricted to a specific index.
Renamed from `IndexEngine` and use metaprogrammation instead of redeclaring each method